source/Calamari.Common/Features/Scripting/Bash/Bootstrap.sh

```diff
@@ -263,4 +263,84 @@ function log_environment_information
 	echo "##octopus[stdout-default]"
 }
 
-log_environment_information
\ No newline at end of file
+log_environment_information
+
+function decrypt_and_parse_variables {
+    local encrypted="$1"
+    local iv="$2"
+
+    local decrypted
+    decrypted=$(decrypt_variable "$encrypted" "$iv")
+    declare -gA octopus_parameters=()
+
+    key_byte_lengths=()
+    value_byte_lengths=()
+    concatenated_hex=""
+    while IFS='$' read -r hex_key hex_value; do
+        hex_value="${hex_value//$'\n'/}"
+        key_byte_len=$(( ${#hex_key} / 2 ))
+        value_byte_len=$(( ${#hex_value} / 2 ))
+        key_byte_lengths+=("$key_byte_len")
+        value_byte_lengths+=("$value_byte_len")
+        concatenated_hex+="${hex_key}${hex_value}"
+    done <<< "$decrypted"
+
+    decoded_output=$(hex_to_string "$concatenated_hex")
+
+    exec 3< <(printf "%s" "$decoded_output")
+
+    for idx in "${!key_byte_lengths[@]}"; do
+        key_byte_len="${key_byte_lengths[idx]}"
+        value_byte_len="${value_byte_lengths[idx]}"
+
+        LC_ALL=C read -r -N "$key_byte_len" decoded_key <&3
+        LC_ALL=C read -r -N "$value_byte_len" decoded_value <&3
+
+        [[ "$decoded_value" == "nul" ]] && decoded_value=""
+        octopus_parameters["$decoded_key"]="$decoded_value"
+    done
+
+    exec 3<&-
+}
+
+hex_to_string() {
```

Comment:
If we use `xxd` this can be removed. xxd is generally included with vim and should be available on most Linux machines but is not generally available on Mac.